contrib/net/http/option.go

```diff
@@ -15,3 +21,36 @@ func WithServiceName(name string) MuxOption {
 		cfg.serviceName = name
 	}
 }
+
+// A RoundTripperBeforeFunc can be used to modify a span before an http
+// RoundTrip is made.
+type RoundTripperBeforeFunc func(*http.Request, ddtrace.Span)
+
+// A RoundTripperAfterFunc can be used to modify a span after an http
+// RoundTrip is made. It is possible for the http Response to be nil.
+type RoundTripperAfterFunc func(*http.Response, ddtrace.Span)
+
+type roundTripperConfig struct {
+	before RoundTripperBeforeFunc
+	after  RoundTripperAfterFunc
+}
+
+// A RoundTripperOption represents an option that can be passed to
+// WrapRoundTripper.
+type RoundTripperOption func(*roundTripperConfig)
+
+// WithRoundTripperBefore adds a RoundTripperBeforeFunc to the RoundTripper
+// config.
+func WithRoundTripperBefore(f RoundTripperBeforeFunc) RoundTripperOption {
+	return func(cfg *roundTripperConfig) {
+		cfg.before = f
+	}
+}
+
```

Comment:
`WithServiceName` already exists for the `MuxOption`

Reply:
I can't create a `WithServiceName` function for `RoundTripperOption` because it's already used by the `MuxOption` which is in the same package. Should I create a `WithRoundTripperServiceName` function?

Reply:
I'm not sure. Would `UseServiceName` be bad?